lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -37,16 +41,38 @@ namespace Js
     {
         Recycler* recycler = scriptContext->GetRecycler();
         SourceTextModuleRecord* childModuleRecord;
-        childModuleRecord = RecyclerNew(recycler, Js::SourceTextModuleRecord, scriptContext);
+        childModuleRecord = RecyclerNewFinalized(recycler, Js::SourceTextModuleRecord, scriptContext);
         // There is no real reference to lifetime management in ecmascript
         // The life time of a module record should be controlled by the module registry as defined in WHATWG module loader spec
         // in practice the modulerecord lifetime should be the same as the scriptcontext so it could be retrieved for the same
-        // site. DOM might hold a reference to the module as well after the initialize the module. 
+        // site. Host might hold a reference to the module as well after initializing the module. 
         // In our implementation, we'll use the moduleId in bytecode to identify the module.
         childModuleRecord->moduleId = scriptContext->GetLibrary()->EnsureModuleRecordList()->Add(childModuleRecord);
         return childModuleRecord;
     }
 
+    void SourceTextModuleRecord::Finalize(bool isShutdown)
```

Comment:
[https://tc39.github.io/ecma262/#sec-source-text-module-records](https://tc39.github.io/ecma262/#sec-source-text-module-records)

---
In reply to: [55075088](https://github.com/Microsoft/ChakraCore/pull/432#discussion_r55075088) [](ancestors = 55075088)